llvm/include/llvm/ADT/PagedVector.h

```diff
@@ -0,0 +1,125 @@
+//===- llvm/ADT/PagedVector.h - 'Lazyly allocated' vectors --------*- C++
+//-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the PagedVector class.
+//
+//===----------------------------------------------------------------------===//
+#ifndef LLVM_ADT_PAGEDVECTOR_H
+#define LLVM_ADT_PAGEDVECTOR_H
+
+#include <vector>
+
+namespace llvm {
+// A vector that allocates memory in pages.
+// Order is kept, but memory is allocated only when one element of the page is
+// accessed. This introduces a level of indirection, but it is useful when you
+// have a sparsely initialised vector like for the case of
+// ASTReader::DeclsLoaded or ASTReader::TypesLoaded.
```

Comment:
Let's avoid using terms like `ASTReader::DeclsLoaded` and `ASTReader::TypesLoaded`. Since I believe most LLVM developers don't and shouldn't understand the properties of these two types.
We should try to describe it directly.

Reply:
Fixed.